resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java

```diff
@@ -65,6 +71,22 @@ public void operationComplete(Future<AddressedEnvelope<DnsResponse, InetSocketAd
                     }
                 }
             };
+    private static final RuntimeException NXDOMAIN_QUERY_FAILED_EXCEPTION = ThrowableUtil.unknownStackTrace(
+            new RuntimeException("No answer found and NXDOMAIN response code returned"),
+            DnsNameResolverContext.class,
+            "onResponse(..)");
+    private static final RuntimeException CNAME_NOT_FOUND_QUERY_FAILED_EXCEPTION = ThrowableUtil.unknownStackTrace(
+            new RuntimeException("No matching CNAME record found"),
+            DnsNameResolverContext.class,
+            "onResponseCNAME(..)");
+    private static final RuntimeException NO_MATCHING_RECORD_QUERY_FAILED_EXCEPTION = ThrowableUtil.unknownStackTrace(
+            new RuntimeException("No matching record type record found"),
+            DnsNameResolverContext.class,
+            "onResponseAorAAAA(..)");
+    private static final RuntimeException UNRECOGNIZED_TYPE_QUERY_FAILED_EXCEPTION = ThrowableUtil.unknownStackTrace(
+            new RuntimeException("Response type was unrecognized"),
+            DnsNameResolverContext.class,
```

Comment:
The type of exception is intentionally not formally defined in the `DnsQueryLifecycleObserver` interface ... so I can change to w/e you prefer. However `DecoderException` javadocs say:

```java
/**
 * An {@link CodecException} which is thrown by a decoder.
 */
```

This is typically not used from a "decoder". What about `DnsNameResolverException` ?